src/game.cpp

```diff
@@ -12298,13 +12240,8 @@ bool game::plmove(int dx, int dy)
         //probability travel through walls but not water
         int tunneldist = 0;
         // tile is impassable
-        while ((m.move_cost(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) == 0) ||
-               // a monster is there
-               ((mon_at(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) != -1 ||
-                 // so keep tunneling
-                 npc_at(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) != -1) &&
-                // assuming we've already started
-                tunneldist > 0)) {
+        tripoint dest( x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy()), u.posz() );
+        while( ( m.move_cost(dest) == 0 || mon_at(dest) != -1 || npc_at(dest) != -1 ) && tunneldist > 0 ) {
```

Comment:
Hello infinite loop, my old friend. This condition only depends on `dest` and `tunneldist` (which is incresed in the loop body, so once it `tunneldist > 0` becomes true, it stays true, until it overflows). But `dest` is not changed in the loop body.

It's not actually infinite as the loop is left when tunneldist reaches 25 or when the character has not enough power left.